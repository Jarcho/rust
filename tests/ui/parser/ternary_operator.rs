// A good chunk of these errors aren't shown to the user, but are still
// required in the test for it to pass.

fn a() {
    let x = 5 > 2 ? true : false;
    //~^ ERROR Rust has no ternary operator
    //~| HELP use an `if-else` expression instead
    //~| ERROR the `?` operator can only be applied to values that implement `Try` [E0277]
    //~| HELP the trait `Try` is not implemented for `{integer}`
    //~| NOTE in this expansion of desugaring of operator `?`
    //~| NOTE the `?` operator cannot be applied to type `{integer}`
    //~| NOTE in this expansion of desugaring of operator `?`
}

fn b() {
    let x = 5 > 2 ? { true } : { false };
    //~^ ERROR Rust has no ternary operator
    //~| HELP use an `if-else` expression instead
    //~| ERROR the `?` operator can only be applied to values that implement `Try` [E0277]
    //~| HELP the trait `Try` is not implemented for `{integer}`
    //~| NOTE in this expansion of desugaring of operator `?`
    //~| NOTE the `?` operator cannot be applied to type `{integer}`
    //~| NOTE in this expansion of desugaring of operator `?`
}

fn c() {
    let x = 5 > 2 ? f32::MAX : f32::MIN;
    //~^ ERROR Rust has no ternary operator
    //~| HELP use an `if-else` expression instead
    //~| ERROR the `?` operator can only be applied to values that implement `Try` [E0277]
    //~| HELP the trait `Try` is not implemented for `{integer}`
    //~| NOTE in this expansion of desugaring of operator `?`
    //~| NOTE the `?` operator cannot be applied to type `{integer}`
    //~| NOTE in this expansion of desugaring of operator `?`
}

fn main() {
    let x = 5 > 2 ? { let x = vec![]: Vec<u16>; x } : { false };
    //~^ ERROR Rust has no ternary operator
    //~| HELP use an `if-else` expression instead
    //~| ERROR expected one of `.`, `;`, `?`, `else`, or an operator, found `:`
    //~| NOTE expected one of `.`, `;`, `?`, `else`, or an operator
    //~| ERROR the `?` operator can only be applied to values that implement `Try` [E0277]
    //~| HELP the trait `Try` is not implemented for `{integer}`
    //~| NOTE type ascription syntax has been removed, see issue #101728 <https://github.com/rust-lang/rust/issues/101728>
    //~| NOTE in this expansion of desugaring of operator `?`
    //~| NOTE the `?` operator cannot be applied to type `{integer}`
    //~| NOTE in this expansion of desugaring of operator `?`
}
