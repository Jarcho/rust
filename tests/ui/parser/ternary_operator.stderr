error: Rust has no ternary operator
  --> $DIR/ternary_operator.rs:5:19
   |
LL |     let x = 5 > 2 ? true : false;
   |                   ^^^^^^^^^^^^^^^
   |
   = help: use an `if-else` expression instead

error: Rust has no ternary operator
  --> $DIR/ternary_operator.rs:16:19
   |
LL |     let x = 5 > 2 ? { true } : { false };
   |                   ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use an `if-else` expression instead

error: Rust has no ternary operator
  --> $DIR/ternary_operator.rs:27:19
   |
LL |     let x = 5 > 2 ? f32::MAX : f32::MIN;
   |                   ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use an `if-else` expression instead

error: expected one of `.`, `;`, `?`, `else`, or an operator, found `:`
  --> $DIR/ternary_operator.rs:38:37
   |
LL |     let x = 5 > 2 ? { let x = vec![]: Vec<u16>; x } : { false };
   |                                     ^ expected one of `.`, `;`, `?`, `else`, or an operator
   |
   = note: type ascription syntax has been removed, see issue #101728 <https://github.com/rust-lang/rust/issues/101728>

error: Rust has no ternary operator
  --> $DIR/ternary_operator.rs:38:19
   |
LL |     let x = 5 > 2 ? { let x = vec![]: Vec<u16>; x } : { false };
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use an `if-else` expression instead

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> $DIR/ternary_operator.rs:5:17
   |
LL |     let x = 5 > 2 ? true : false;
   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
   |
   = help: the trait `Try` is not implemented for `{integer}`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> $DIR/ternary_operator.rs:16:17
   |
LL |     let x = 5 > 2 ? { true } : { false };
   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
   |
   = help: the trait `Try` is not implemented for `{integer}`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> $DIR/ternary_operator.rs:27:17
   |
LL |     let x = 5 > 2 ? f32::MAX : f32::MIN;
   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
   |
   = help: the trait `Try` is not implemented for `{integer}`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> $DIR/ternary_operator.rs:38:17
   |
LL |     let x = 5 > 2 ? { let x = vec![]: Vec<u16>; x } : { false };
   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
   |
   = help: the trait `Try` is not implemented for `{integer}`

error: aborting due to 9 previous errors

For more information about this error, try `rustc --explain E0277`.
